src/sentry/contexts.cpp

```diff
@@ -0,0 +1,340 @@
+#include "contexts.h"
+
+#include "sentry/environment.h"
+#include "sentry/uuid.h"
+
+#include <godot_cpp/classes/dir_access.hpp>
+#include <godot_cpp/classes/display_server.hpp>
+#include <godot_cpp/classes/engine.hpp>
+#include <godot_cpp/classes/os.hpp>
+#include <godot_cpp/classes/performance.hpp>
+#include <godot_cpp/classes/project_settings.hpp>
+#include <godot_cpp/classes/rendering_server.hpp>
+#include <godot_cpp/classes/time.hpp>
+
+namespace {
+
+String _screen_orientation_as_string(int32_t p_screen) {
+	ERR_FAIL_NULL_V(DisplayServer::get_singleton(), "");
+	switch (DisplayServer::get_singleton()->screen_get_orientation(p_screen)) {
+		case DisplayServer::SCREEN_LANDSCAPE: {
+			return "Landscape";
+		} break;
+		case DisplayServer::SCREEN_PORTRAIT: {
+			return "Portrait";
+		} break;
+		case DisplayServer::SCREEN_REVERSE_LANDSCAPE: {
+			return "Landscape (reverse)";
+		} break;
+		case DisplayServer::SCREEN_REVERSE_PORTRAIT: {
+			return "Portrait (reverse)";
+		} break;
+		case DisplayServer::SCREEN_SENSOR_LANDSCAPE: {
+			return "Landscape (defined by sensor)";
```

Comment:
As per [DisplayServer doc](https://docs.godotengine.org/en/stable/classes/class_displayserver.html#enum-displayserver-screenorientation), `SCREEN_SENSOR_LANDSCAPE` can be either default or reversed, based on the sensor.

Reply:
Sometimes you can force an orientation, like lock the screen in the landscape mode. The difference is that there are two landscape orientations: default and reversed (up-side down?). And the "Landscape (defined by sensor)" here means the appropriate one is chosen automatically based on the sensors' data. At least, these are the enum values reported by the engine. Maybe we can use better string representations for these?

Reply:
Yeah just wondering if it's worth adding the "sensor" one since it adds complexity. We could just map both to the same "Landscape", regardless if it's sensor one or not

Reply:
@bruno-garcia 
I can make a PR. Technically, there's 3 different landscape values. What about "Landscape (reverese)"? Keeping or grouping it all under "Landscape"?